RavenDB-23464 Validate during index compilation that the index is using Corax as the search engine when the map contains `CreateVector`.

maciejaszyk opened the pull request:
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-23464

### Additional description


### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [x] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

maciejaszyk committed:
RavenDB-23464 Validate during index compilation that the index is using Corax as the search engine when the map contains `CreateVector`.

karmeli87 commented:
And if we have `CreateVector` in the Reduce part? would it also fail on compilation?

maciejaszyk committed:
RavenDB-23464
- Added the map-reduce test
- Added the link to documentation about search engine in exception.

maciejaszyk commented:
> And if we have `CreateVector` in the Reduce part? would it also fail on compilation?

added
